Failing tests when run locally on Windows

**Describe the bug**

Some tests fail on Windows (Win 11 Pro, Python 3.12.5). The requests_cache SQLite file cannot be removed at the end of the tests.

**Version of LinkML you are using**

1.8.2

The tests don't fail in CI because (a) tests are just run once and (b) git is configured in the "right way".

```cmd
C:\Users\dlinke\MyProg_local\gh-dalito\linkml (main -> origin)
(.venv) λ pytest
===================================== short test summary info =====================================
FAILED tests/test_base/test_markdown.py::test_models_markdown[C:\\Users\\dlinke\\MyProg_local\\gh-dalito\\linkml\\.venv\\Lib\\site-packages\\linkml_runtime\\linkml_model\\model\\schema\\types.yaml] - AssertionError: assert value matches snapshot C:\Users\dlinke\MyProg_local\gh-dalito\linkml\tes...
FAILED tests/test_base/test_markdown.py::test_models_markdown[C:\\Users\\dlinke\\MyProg_local\\gh-dalito\\linkml\\.venv\\Lib\\site-packages\\linkml_runtime\\linkml_model\\model\\schema\\mappings.yaml] - AssertionError: assert value matches snapshot C:\Users\dlinke\MyProg_local\gh-dalito\linkml\tes...
FAILED tests/test_base/test_markdown.py::test_models_markdown[C:\\Users\\dlinke\\MyProg_local\\gh-dalito\\linkml\\.venv\\Lib\\site-packages\\linkml_runtime\\linkml_model\\model\\schema\\annotations.yaml] - AssertionError: assert value matches snapshot C:\Users\dlinke\MyProg_local\gh-dalito\linkml\tes...
FAILED tests/test_base/test_markdown.py::test_models_markdown[C:\\Users\\dlinke\\MyProg_local\\gh-dalito\\linkml\\.venv\\Lib\\site-packages\\linkml_runtime\\linkml_model\\model\\schema\\extensions.yaml] - AssertionError: assert value matches snapshot C:\Users\dlinke\MyProg_local\gh-dalito\linkml\tes...
FAILED tests/test_issues/test_issue_179.py::test_issue_179 - AssertionError: assert value matches snapshot C:\Users\dlinke\MyProg_local\gh-dalito\linkml\tes...
FAILED tests/test_issues/test_issue_62.py::test_issue_62 - AssertionError: assert value matches snapshot C:\Users\dlinke\MyProg_local\gh-dalito\linkml\tes...
FAILED tests/test_issues/test_issue_65.py::test_issue_65 - AssertionError: assert value matches snapshot C:\Users\dlinke\MyProg_local\gh-dalito\linkml\tes...
FAILED tests/test_scripts/test_gen_golr.py::test_metamodel_valid_call - AssertionError: assert value matches snapshot C:\Users\dlinke\MyProg_local\gh-dalito\linkml\tes...
FAILED tests/test_utils/test_deprecation.py::test_deprecation_str - AssertionError: assert '[test-dep] \...nkml/issues/1' == '[test-dep] R...nkml/issues/1'
= 9 failed, 7340 passed, 1290 skipped, 10 xfailed, 301 warnings, 261 subtests passed in 382.83s (0:06:22) =
```

Details on the failing removal of requests_cache file:
<details>

```cmd
C:\Users\dlinke\MyProg_local\gh-dalito\linkml (main -> origin)
(.venv) λ pytest -k deprecat
======================================= test session starts =======================================
platform win32 -- Python 3.12.5, pytest-7.4.4, pluggy-1.5.0
rootdir: C:\Users\dlinke\MyProg_local\gh-dalito\linkml
configfile: pyproject.toml
plugins: subtests-0.11.0
collected 8729 items / 8689 deselected / 40 selected

tests\test_compliance\test_metadata_compliance.py ssssssssss                                 [ 25%]
tests\test_utils\test_deprecation.py ............................                            [ 95%]
tests\test_validator\test_cli.py .                                                           [ 97%]
tests\test_utils\test_deprecation.py .E                                                      [100%]

============================================= ERRORS ==============================================
__________________________ ERROR at teardown of test_removed_are_removed __________________________

pytestconfig = <_pytest.config.Config object at 0x000001D8B15F5940>

    @pytest.fixture(scope="session", autouse=True)
    def patch_requests_cache(pytestconfig):
        """
        Cache network requests - for each unique network request, store it in
        an sqlite cache. only do unique requests once per session.
        """
        if pytestconfig.getoption("--without-cache"):
            yield
            return
        cache_file = Path(__file__).parent / "output" / "requests-cache.sqlite"
        requests_cache.install_cache(
            str(cache_file),
            backend="sqlite",
            urls_expire_after={"localhost": requests_cache.DO_NOT_CACHE},
        )
        requests_cache.clear()
        yield
        # delete cache file unless we have requested it to persist for inspection
        if not pytestconfig.getoption("--with-output"):
>           cache_file.unlink(missing_ok=True)

tests\conftest.py:244:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = WindowsPath('C:/Users/dlinke/MyProg_local/gh-dalito/linkml/tests/output/requests-cache.sqlite')
missing_ok = True

    def unlink(self, missing_ok=False):
        """
        Remove this file or link.
        If the path is a directory, use rmdir() instead.
        """
        try:
>           os.unlink(self)
E           PermissionError: [WinError 32] Der Prozess kann nicht auf die Datei zugreifen, da sie von einem anderen Prozess verwendet wird: 'C:\\Users\\dlinke\\MyProg_local\\gh-dalito\\linkml\\tests\\output\\requests-cache.sqlite'

C:\dev\Python312\Lib\pathlib.py:1342: PermissionError
===================================== short test summary info =====================================
ERROR tests/test_utils/test_deprecation.py::test_removed_are_removed - PermissionError: [WinError 32] Der Prozess kann nicht auf die Datei zugreifen, da sie von einem...
============== 30 passed, 10 skipped, 8689 deselected, 4 warnings, 1 error in 7.24s ===============
```
</details>

--- a/linkml/generators/pydanticgen/pydanticgen.py
+++ b/linkml/generators/pydanticgen/pydanticgen.py
@@ -1061,7 +1061,7 @@ class PydanticGenerator(OOCodeGenerator, LifecycleMixin):
         # interpret all imported schema paths as relative to that
         output_path.parent.mkdir(parents=True, exist_ok=True)
         serialized = generator.serialize(rendered_module=rendered)
-        with open(output_path, "w") as ofile:
+        with open(output_path, "w", encoding="utf-8") as ofile:
             ofile.write(serialized)
 
         results.append(
@@ -1080,7 +1080,7 @@ class PydanticGenerator(OOCodeGenerator, LifecycleMixin):
             rel_path = _import_to_path(generated_import.module)
             abs_path = (output_path.parent / rel_path).resolve()
             abs_path.parent.mkdir(parents=True, exist_ok=True)
-            with open(abs_path, "w") as ofile:
+            with open(abs_path, "w", encoding="utf-8") as ofile:
                 ofile.write(serialized)
 
             results.append(
@@ -1123,7 +1123,7 @@ def _ensure_inits(paths: List[Path]):
     common_path = Path(os.path.commonpath(paths))
 
     if not (ipath := (common_path / "__init__.py")).exists():
-        with open(ipath, "w") as ifile:
+        with open(ipath, "w", encoding="utf-8") as ifile:
             ifile.write(" \n")
 
     for path in paths:
@@ -1131,7 +1131,7 @@ def _ensure_inits(paths: List[Path]):
         path = path.parent
         while path != common_path:
             if not (ipath := (path / "__init__.py")).exists():
-                with open(ipath, "w") as ifile:
+                with open(ipath, "w", encoding="utf-8") as ifile:
                     ifile.write(" \n")
             path = path.parent
 

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -238,7 +238,9 @@ def patch_requests_cache(pytestconfig):
         urls_expire_after={"localhost": requests_cache.DO_NOT_CACHE},
     )
     requests_cache.clear()
-    yield
+    with requests_cache.enabled():
+        yield
+    requests_cache.uninstall_cache()
     # delete cache file unless we have requested it to persist for inspection
     if not pytestconfig.getoption("--with-output"):
         cache_file.unlink(missing_ok=True)

--- a/tests/test_notebooks/test_examples.py
+++ b/tests/test_notebooks/test_examples.py
@@ -17,7 +17,7 @@ class NotebookTests(unittest.TestCase):
         output_file.parent.mkdir(parents=True, exist_ok=True)
         with redirect_stdout(output):
             importlib.import_module(import_module)
-        with open(output_file, "w") as f:
+        with open(output_file, "w", encoding="utf-8") as f:
             f.write(
                 re.sub(
                     r'Generation date: .*?(["\n])',

--- a/tests/test_notebooks/test_notebooks.py
+++ b/tests/test_notebooks/test_notebooks.py
@@ -43,7 +43,7 @@ class NotebookTestCase(TestEnvironmentTestCase):
             )
             and self.env.fail_on_error
         ):
-            with open(nbenv.temp_file_path(nbname), "w") as actualf:
+            with open(nbenv.temp_file_path(nbname), "w", encoding="utf-8") as actualf:
                 actualf.write(outf.getvalue())
 
     def test_redo_notebooks(self):

--- a/tests/test_utils/test_deprecation.py
+++ b/tests/test_utils/test_deprecation.py
@@ -177,7 +177,7 @@ def test_deprecation_props(deprecated, deprecated_val, removed, removed_val, lin
     if removed is not None:
         remver = SemVer(major=linkml_version.major + removed)
 
-    dep = Deprecation(name="test-dep", message="testing our properites", deprecated_in=depver, removed_in=remver)
+    dep = Deprecation(name="test-dep", message="testing our properties", deprecated_in=depver, removed_in=remver)
     assert dep.deprecated == deprecated_val
     assert dep.removed == removed_val
 

